src/openloco/ui.cpp

```diff
@@ -818,10 +918,13 @@ namespace openloco::ui
 
         // Update config fullscreen resolution if not set
         auto& cfg = config::get();
-        if (cfg.resolution_width == std::numeric_limits<uint16_t>::max() && cfg.resolution_height == std::numeric_limits<uint16_t>::max())
+        auto& cfg_new = config::get_new();
+        if (!(cfg_new.display.fullscreen_resolution.isPositive() && cfg.resolution_width > 0 && cfg.resolution_height > 0))
```

Comment:
This has changed the logic from the original; the check for cfg.resolution_width == cfg.resolution_height == -1 (65535) is no longer present.